### DISCUSSION TITLE
RDF processing and Knowledge Graph ER

### DISCUSSION THREAD

**Discussion by reversingentropy** [AI-team-UoA/pyJedAI#9] [Q&A] [answered]
Hi,

First off, thank you so much for developing this entity resolution repository!

I am particularly interested in performing dirty entity resolution on a knowledge graph comprising multiple subgraphs. Therefore, it is a dirty multi source ER scenario. 

1) However, converting triples via RDFLib from a large knowledge graph into a DataFrame can consume a significant amount of time and computational resources. Hence, I'm reaching out to inquire if you have any recommendations or suggestions to optimize this process.

2) Furthermore, if these are knowledge graph triples, can we treat URIs as IDs ? Furthermore, if URIs are linked together via a relationship or feature, how will pyJedAI take this information into account, instead of string matching these URIs?

Once again, thank you for your valuable work!!!

**Comment by Nikoletos-K** [comment, marked as answer]
Hello!  👋

We can only process RDF data as strings at the moment. This manner, we may make use of all of our workflows. Converting triples from a large knowledge network into a DataFrame via RDFLib may certainly be time intensive. But, we intend to improve pyJedAI with more semantic web functionality. For your second question, the triples are converted to strings, and the URIs you indicate are the IDs that show the matched pairs.

Thank you!

**Reply by reversingentropy** [reply]
Hi Nikoletos,
Thank you again for your swift reply! 
I am curious, could you elaborate on the semantic web functionality? 
Are there plans to employ the use of knowledge graph embeddings via [pykeen](https://pykeen.readthedocs.io/en/stable/)?

**Reply by Nikoletos-K** [reply]
We're actively integrating several [JedAI-Spatial](https://github.com/AI-team-UoA/JedAI-spatial) functionalities to pyJedAI. However, it is not a top priority right now.

pykeen looks interesting, we'll check it out!

**Reply by reversingentropy** [reply]
Thank you Nikoletos, I will check JedAI-Spatial. Have a good day!
